Bottom: c5bd0b81f3e56159d325a97a21cc0d306515e9ec
Top:    215f29bb14058afa99213c595556e70146bc456c
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-03-22 14:22:02 +0800

Refresh of vhost_net-prefetch-desc

---

diff --git a/drivers/vhost/net.c b/drivers/vhost/net.c
index 74d5961..31199b2 100644
--- a/drivers/vhost/net.c
+++ b/drivers/vhost/net.c
@@ -702,6 +702,7 @@ static void handle_rx_batched(struct vhost_net *net)
 					    cpu_to_vhost32(vq, len), i);
 		}
 		while (nvq->rh != nvq->rt) {
+			
 	}
 }
 
diff --git a/drivers/vhost/vhost.c b/drivers/vhost/vhost.c
index ed90154..747ac7e 100644
--- a/drivers/vhost/vhost.c
+++ b/drivers/vhost/vhost.c
@@ -1996,7 +1996,7 @@ static int get_indirect(struct vhost_virtqueue *vq,
 
 /* Prefetch descriptor indices */
 int vhost_prefetch_desc_indices(struct vhost_virtqueue *vq,
-				struct vring_used_elem_heads *heads, u16 num)
+				__virtio16 *indices, u16 num)
 {
 	int ret = 0;
 	u16 last_avail_idx, total;
@@ -2013,14 +2013,14 @@ int vhost_prefetch_desc_indices(struct vhost_virtqueue *vq,
 	ret = total = min(total, num);
 
 	while (total) {
-		int ret2 = vhost_get_avail(vq, avail_idx,
+		int ret2 = vhost_get_avail(vq, indices[0],
 			   &vq->avail->ring[last_avail_idx & (vq->num - 1)]);
 		if (unlikely(ret2)) {
 			vq_err(vq, "Failed to get descriptors\n");
 			return -EFAULT;
 		}
 		--total;
-		heads[indices++].id = vhost16_to_cpu(vq, avail_idx);
+		++indices;
 		++last_avail_idx;
 	}
 	/* FIXME: update used ring here? together with batch dequing? */
